File: caesar.pas
{ Jack Hueber }
program Caesar;
{global variable for the string}
var str: String;

function encrypt(str: String; shift: Integer): String;
	var
		i, temp: Integer;
		capStr: String;
	begin
		capStr := UpCase(str);
		for i := 1 to length(capStr) do
		begin
			{ Here we make this shift change by setting it to the integer temp }
			temp := (byte(capStr[i]) - 65 + shift) mod integer(26);
			if (temp < 0) then
				begin
					temp := temp + 26;
				end;
			temp := temp + 65;
			{ Change letter in the str for position i }
			if (byte(capStr[i]) = 32) then
				begin
					temp := 32;
				end;
			capStr[i] := chr(temp);
		end;
		encrypt := capStr;
	end;

function decrypt(str: String; shift: Integer): String;
	var
		i, temp: Integer;
		capStr: String;
	begin
		capStr := UpCase(str);
		for i := 1 to length(capStr) do
		begin
			{ Here we make this shift change by setting it to the integer temp.
				Also subtract the shift now }
			temp := (byte(capStr[i]) - 65 - shift) mod integer(26);
			if (temp < 0) then
				begin
					temp := temp + 26;
				end;
			temp := temp + 65;
			{ Change letter in the str for position i }
			if (byte(capStr[i]) = 32) then
				begin
					temp := 32;
				end;
			capStr[i] := chr(temp);
		end;
		decrypt := capStr;
	end;

procedure solve(str: String; fullShift: Integer);

var
	i: Integer;
begin
	for i := 0 to fullShift do
	begin
	{ Encrpyt is called the fullShift amount, i increments to the fullshift amount }
		Writeln('Caesar ', i, ': ', encrypt(str, i));
	end;
end;

{ Now we call all of the functions, the shift amount being 1 for encrypt and decrypt
	Solve has 26 so it does encrypt for 26 times}
begin
	str := 'hal';
	str := encrypt(str, 1);
	Writeln(str);
	str := decrypt(str, 1);
	Writeln(str);
	solve(str, 26);
end.

File: caesar.scala
// Jack Hueber
object Main extends App {
	class Caesar {

	// Global varible str as a string
	var str = ""

	// shift is how many letters we shift each position, we return the shifted string
	def encrypt(shift: Int): String = {
		val newShift = shift % 26
		
		//change to uppercase to make shifting easier
		val strArray = this.str.toUpperCase().toCharArray()
		var retString = ""

		for (i<-strArray) {
			if (i.toInt == 32) {
				retString += " ";
			} else {
				var temp = (i.toInt - 65 + newShift) % 26
				if (temp < 0) {
					temp += 26
				}
				retString += (temp + 65).toChar
			}
		}

		return retString
	}

	// shift is how many letters we shift each position, we return the shifted string
	// we substract the shift rather than add to be the opposite of encrypt
	def decrypt(shift: Int): String = {
		val newShift = shift % 26
		
		// change to upper case to make shifting easier
		val strArray = this.str.toUpperCase().toCharArray()
		var retString = ""

		for (i<-strArray) {
			if (i.toInt == 32) {
				retString += " ";
			} else {
				var temp = (i.toInt - 65 - newShift) % 26
				if (temp < 0) {
					temp += 26
				}
				retString += (temp + 65).toChar
			}
		}

		return retString
	}

	// full shift is the amount of times we loop through encrypt, we return the list of encrypted strings
	def solve(fullShift: Int) {
		for (i <- 0 to fullShift) {
			println("Caesar " + i + ": " + encrypt(i))
		}
	}
}
	var caesar = new Caesar()
		caesar.str = "HAL"
		caesar.str = caesar.encrypt(1)
		println(caesar.str)
		caesar.str = caesar.decrypt(1)
		println(caesar.str)
		println(caesar.solve(26))
}